### ISSUE TITLE
Moka pot: use beverage quantity not amount of water for calculation of brew ratio

### ISSUE THREAD

**Issue by Jan-mjm** [graphefruit/Beanconqueror#464]
In moka pots, a lot of water always remains in the bottom part and dos not affect the brew at all. Therefore it only makes sense to use _beverage quantity_ for calculation the _brew ratio_.

For me, it's not a big deal of entering the actual _beverage quantity_ into the field for _amount of water_.
However, this
a) could be misleading to others if the y do not spot this behaviour 
b) would be very unfortunate _if_ the the formular used for calculating the brew ratio get's changed at some point, making a lot of false calculations for old data.

If you already have an opinion on this, it would be very nice to hear your intention.

Other _Preparation Style: Percolators_ work different and require the _amount of water_ to be used. So maybe creating _Preparation Style: Moka Pod_ could be the best option. Alternatively you could let the user decide which one to use for each _preparation method_.

**Comment by graphefruit** [comment]
Hey!
Actually I didn't do any thoughts about this.
This use case is actually very niche, and I don't know if there are more edge cases like the mocca pot.
Are there other brewers out there which will result with this issue or have a different calculated brew ratio?

**Comment by Jan-mjm** [comment]
While moka pods aren't that niche at all, using your app to log data for one most certainly is. ;)
I'm not aware of any other brewing method where you have to use _beverage quantity_ calculate the brew ratio (except real espresso of course).
However, I see an increasing amount of people doing it this way for filter coffee as well. It is supposed to increase consistency across different coffee beans, that absorb different amounts of water. (I'm not yet sure about significance.)

For now, I just chose to use a customised espresso _brewing method_ for my moka pod. The _amount of water_ does not matter if it's enough but below the valve. This just ensures data consistency for me ...

But I can imagine other users, not realising which input field is the import one. I just realised it after a couple of days when I played with the extraction time but the brew ratio didn't change at all.

Either way, thank you for your great work. :)

**Comment by gudvinr** [comment]
It makes no sense to do it this way from value standpoint.
You don't know beforehand how much water you'll end up with because it depends on grind size, temperature, heating intensity, brew time, etc.

So if you tie brew ratio to _output_ value, you end up with information you can't use to make another brew.  
But brew ratio for _input_ value is consistent and can be reused.

However, what will be interesting is to add output to input water ratio that is calculated automatically. E.g. if you put 300ml of water and end up with 200ml of beverage, it'll show you 67%.

**Comment by Jan-mjm** [comment]
> It makes no sense to do it this way from value standpoint.
I wouldn't say that at all from a scientific standpoint.
Since, if done right, there is always quite an amount of water remaining in the bottom chamber, that does not get in contact with the coffee grounds and therefore is not part of brewing process itself. It would just be scientifically wrong and confusing to call the ratio of _grounds_ and _total water_ a _brew ratio_. The _brew ratio_ is an inherent property of the coffee produced, but the ratio of _grounds_ and _total water_ a _brew ratio_ needed to get a certain result is an inherent property of your whole experimental setup including all other parameters and the exakt model of moka pod you use.

Myself, I have do different sized moka pods of the same product line. The smaller one has relatively more left over water since the base is relatively larger and the riser pipe ends at approx. the same distance above the bottom.
If I aim for an _actual brew ratio_ of 1:6.2, I end up with two results tasting the same. If I calculate the ratio of _grounds_ and _total water_ the results a _way_ off: 1:12.5 and 1:9.8 for the small and the larger one respectively. Therefore interchanging those numbers with different people using different moka pods would be useless.

> You don't know beforehand how much water you'll end up with because it depends on grind size, temperature, heating intensity, brew time, etc.
Yes, that's right. Knowing the _total amount of water_ helps with repeatability. It is definitely helpful to log the _total amount of water_ as well for a certain model of moka pot. But it should not and does not need to be used to _calculate a brew ratio_.
It's sufficient for repeatability to write down the _total amount of water. For this, I use the comment field at the bottom as it does not make any sense to calculate something with this value.
So interestingly, my argument is quite similar to yours: reproducibility, not to confuse with your argument of repeatability. 
 
> So if you tie brew ratio to _output_ value, you end up with information you can't use to make another brew.
It's quite easy to get to a certain _actual brew ratio_ using any moka pod. All you need is two brews. For the first, you use the maximum amount of total water (right up to the valve) weight it, then after the brew weight the coffee and calculate the _actual brew ratio_. Then you can use, let's say 15 grams less total water and you will receive 15 g less coffee. (This only relationship is only linear in a certain range, but if you use too less water the moka pods won't work anyway.)

> But brew ratio for _input_ value is consistent and can be reused.
Only for a certain model of moka pod, as pointed out above. Reproducibility is very bad and you end up calling something a brew ratio that's not one at all: The ratio of _grounds_ and _total water_ does not correlate with strength of coffee if different moka pods are used.

> However, what will be interesting is to add output to input water ratio that is calculated automatically. E.g. if you put 300ml of water and end up with 200ml of beverage, it'll show you 67%.
I wouldn't advice that at all. Sounds very confusing as you are changing your definition of _brew ratio_ depending on what data you measured. From a data standpoint, this makes things very messy for later data processing as the different definitions easily lead to differences by factor 2 (!) or more.

**Comment by gudvinr** [comment]
Main point of brew ratio in first place is to be able to reproduce it. You write it down and just use it.  
If you define brew ratio in terms of output values it is not reproducible, thus useless from practical standpoint and basically only shows you some arbitrary numbers.

**Comment by Jan-mjm** [comment]
> Main point of brew ratio in first place is to be able to reproduce it. You write it down and just use it. If you define brew ratio in terms of output values it is not reproducible, thus useless from practical standpoint and basically only shows you some arbitrary numbers.

No, that's just incorrect, as I pointed out above and elaborated below.
Using the real _brew ratio_ **is** more practical and definitely not arbitrary.

> Last thing everyone wants in the morning is to crunch numbers to estimate amount of water they need to end up with some output amount of water that will give you _the true brew ratio_.

There is no need to crunch numbers in the morning at all. 
Below, there is a table of real numbers from two weeks ago. The coffee was the same and there cups tastet very much the same. As you can see, this is because of the (almost) same brew ratios. The fractions of _coffee grounds /  total water_ are totally different, have nothing to do with the taste and are not needed for making the coffee.

|   Model   |   _coffee grounds_ / g  |   _total water_ / g  |   _brewed coffee_ / g  |   _brew ratio_  |   _coffee grounds /  total water_  |
|-----------|-----------------------|--------------------|----------------------|---------------|----------------------------------|
|   1 cup   |   8.4                 |   210              |   52.1               |   1:6.2       |   1:25                           |
|   3 cups  |   19.0                |   340              |   120.4                |   1:6.3       |   1:17                           |


Last week, I had a new coffee, that was a bit weaker required a lower _brew ratio_. This was great:

|   Model   |   _coffee grounds_ / g  |   _total water_ / g  |   _brewed coffee_ / g  |   _brew ratio_  |   _coffee grounds /  total water_  |
-- | -- | -- | -- | -- | --
1 cup | 8.4 | 200 | 40.9 | 1:4.9 | 1:24

Now, what do I have to do, too get the same result in the larger moka pot? Right, I want to decrease the brew ratio by a little bit more than 1. I can easily reduce the brew ratio by 1, if I reduce the amount of brewed coffee exactly by the mass of ground coffee I use, which is 19 g. Since I would like to reduce the _brew ratio_ a little bit more, I want 20 g less coffee brewed. 
As the amount of remaining water in the bottom chamber is about constant, I just add 20 g less _total water_. Can't be easier than that.
The result was:

|   Model   |   _coffee grounds_ / g  |   _total water_ / g  |   _brewed coffee_ / g  |   _brew ratio_  |   _coffee grounds /  total water_  |
-- | -- | -- | -- | -- | --
3 cups | 19.0 | 320 | 98.2 | 1:5.2 | 1:16.8

It did taste like the one with the new coffee prepared in the smaller moka pod.

As you can see your proposed ratios of _coffee grounds /  total water_ are in fact quite arbitrary as they neither represent the taste of the cup, nor are useful while making the coffee. The _coffee grounds /  total water_ isn't useful at all for making coffee, but the _total water_ and the true _brew ratio_ are the ones you need. 
If you would be working with _coffee grounds /  total water_, THAT would involve the number crunching nobody likes in the morning.
The very last thing everybody wants is the need to remember different brew ratios for different moka pods.

Another upside of the real _brew ratios_ is, that you can talk with other people using different equipment, where the residue of water can be quite different depending of the geometry and size of their moka pod.

**Comment by gudvinr** [comment]
> Using the real brew ratio is more practical and definitely not arbitrary.

No, it is completely impractical because you can't use brew ratio based on output value to control input since you basically introduce other variables in brew ratio. You can't possibly know how much water you need to add to get target brew ratio, thus it is only valid for this particular brew.

Extraction will differ from ground to ground, it will differ depending on puck thickness, coffee density, whether or not you use filter, etc. It will change for same grounds if you use different water temperature and/or heating temperature because it will change point where water starts to actively boil inside, which introduces air bubbles that basically mean you should cut off the heat.

Sure, you can't use these ratios exactly for different pots but if you just have one, it does not matter.
If you do, however, need to use different brewers, you are saying yourself:
> As the amount of remaining water in the bottom chamber is about constant, I just add 20 g less total water.

Since you already do that, you can just write down "total water - 20 g", thus making ratios both constant and independent of other variables.

**Comment by Jan-mjm** [comment]
Okay, it seems like you're seeing the brew ratio as a completely different thing with a different purpose.
For you it needs just needs to be exactly known _ex ante_ in order how much water you will use. I believe, the use of the parameter _total water_ ist perfectly sufficient for that.
I see the _brew ratio_ as a measure of strength for the coffee, generally it correlates with the % of TDS.
And yes, it correlates with a whole lot of other parameters. This is in fact no problem as you don't change a lot of parameters at the same time. Usually it's sufficient to either change _dose_, _grind setting_ or _total water_.
I don't need to now the _brew ratio_ for my daily coffee. However, it comes important, when I want to change the strength/dilution of my coffee by a certain factor, want to use another pod or talk to other people. _Brew ratio_ is being established for that.

> No, it is completely impractical because you can't use brew ratio based on output value to control input since you basically introduce other variables in brew ratio.

Seriously, it's not. I have shown the practicality in the example above. It does work in practice. 

> Extraction will differ from ground to ground, it will differ depending on puck thickness, coffee density, whether or not you use filter, etc. It will change for same grounds if you use different water temperature and/or heating temperature because it will change point where water starts to actively boil inside, which introduces air bubbles that basically mean you should cut off the heat.

Of course, that's all true. Many of the things you are mentioning only cause marginal statistical differences that can be neglected. If you use the same (or even a similar coffee), with the same dose, grind setting and brew ratio, physical quantities like puck thickness or coffee density do not have a relevant impact on the result. Off course, I'm not talking about using a paper filter etc. If you use one, this has a significant influence on taste so that you need to optimise the real _brew ratio_ again.

Can you please provide an example on how you use your concept of "brew ratio" and why you can't just solely work with the parameter of _total water_ if you're only using one singular Moka pod anyway?

> Since you already do that, you can just write down "total water - 20 g", thus making ratios both constant and independent of other variables.

Not exactly. Having a fixed 20 g would not be universally valid. In his case, one could say "total water - once the mass of ground coffee". Which in fact is _by definition_ the equivalent of reducing the brew ratio by 1.

**Comment by gudvinr** [comment]
> why you can't just solely work with the parameter of total water if you're only using one singular Moka pod anyway?

I can... That's what I do basically. I just look at brew ratio I had last time (grounds/total water) and add whatever amount of water I need for amount of grounds I made. Although I usually don't change amount of grounds at all.

I know that in this case you still introduce errors because actual brew ratio will be grounds/(total water - leftover) and if you change amount of grounds, you need to actually add a bit more water.  
But since you can't control amount of leftover water it doesn't matter because it actually varies enough to make this error insignificant. At least until you try to make 3cup brew in 6cup moka pot.

> I don't need to now the brew ratio for my daily coffee. 

That was kinda my point. I have nothing against this to be used as statistical data or e.g. secondary value (even for every brew type, really), but you were suggesting using beverage quantity to be used _instead_ of total water. 

> Seriously, it's not. I have shown the practicality in the example above. It does work in practice.

I didn't mean it in a way that you can't use this data for something (at all). It has a value and I see your point.  
I was just saying that when you get up in the morning and make a cuppa in your pajamas, these values won't really help you to get a good morning brew until you do some post processing to make them make sense.

P.S. It's "pot", not a "pod"

**Comment by mamartins** [comment]
I fully agree with the original post’s point. It doesn’t make sense to include water that was not used in the final brew when calculating the brew ratio.

Now, I do think that both inputs should be kept as by keeping them, we can accurately track the actual water used and the initial amount of water added as there is a direct connection between then. This approach allows for more precise monitoring of any changes to the input and the actual brewing process. It also provides more detailed data for analysis, which could be beneficial for improving brew consistency and quality over time.
